xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3764,6 +3776,13 @@ bool CVideoPlayer::CloseStream(CCurrentStream& current, bool bWaitForBuffers)
   if(bWaitForBuffers)
     SetCaching(CACHESTATE_DONE);
 
+  if (STREAM_SOURCE_MASK(current.source) == STREAM_SOURCE_DEMUX)
+  {
+    if (!m_pDemuxer)
+      return false;
```

Comment:
Closing a stream player and disabling a demux steam are two different stories. Check out this sequence: https://github.com/mapfau/xbmc/blob/inputstream.pts/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L1433
It deletes the demuxer and then closes stream players.